cli/azd/cmd/deploy.go

```diff
@@ -221,7 +235,14 @@ func (da *deployAction) Run(ctx context.Context) (*actions.ActionResult, error)
 			continue
 		}
 
-		deployTask := da.serviceManager.Deploy(ctx, svc, nil)
+		var packageOutput *project.ServicePackageResult
+		if da.flags.fromPackage != "" {
+			packageOutput = &project.ServicePackageResult{
+				PackagePath: da.flags.fromPackage,
```

Comment:
It feels like `azd` should have some `validation` before using the input path/tag and invoking deploy.  it seems like a security vulnerability.
Should azd take a package that was created by azd only?  

What happen, for example, if I call `azd deploy service-name --from-package` and I use a path that is nothing related to the `service-name` ( for example, the service in the template `foo` is a python app and I pass the path to a node app ).  Would that be an error?  or should azd just follow the order and deploy? (as per customer's request)?

Reply:
Azd will not validate that the package was generated by azd or that it is the correct service / language.  It is up to the user to ensure they are passing a compatible package.

This supports scenarios where devs can build once, deploy multiple times promoting application package builds up through dev, test, prod environments.

The service target is responsible for ensuring the `--from-package` parameter is of the expected type.  Ex) app service will fail if not provided a valid zip file path.

Container apps will fail if the value is not a valid local docker image tag.